CM7/Core/Src/gatedriver.c

```diff
@@ -0,0 +1,93 @@
+
+#include "gatedriver.h"
+#include "stm32h7xx.h"
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+
+//TODO: Look up STM callback func pointer for ADCs
+static void gatedrv_current_adc_cb(gatedriver_t* drv)
+{
+
+}
+
+static void gatedrv_voltage_adc_cb(gatedriver_t* drv)
+{
+
+}
+
+static void gatedrv_ready_cb(gatedriver_t* drv)
+{
+
+}
+
+static void gatedrv_reset_cb(gatedriver_t* drv)
+{
+
+}
+
+static void gatedrv_fault_cb(gatedriver_t* drv)
+{
+
+}
+
+gatedriver_t* init_gatedriver()
```

Comment:
The signature in the header has the identifier as `init_gatedrv` - mismatch here